### PR TITLE
feat(policy-templates): add Route53ChangeResourceRecordSetsPolicy

### PR DESCRIPTION
**Pull request by theBaffo**
*Issue #, if available:* https://github.com/aws/serverless-application-model/issues/1930

*Description of changes:* Add "Route53ChangeResourceRecordSetsPolicy" policy

*Description of how you validated changes:*

*Checklist:*

- [x] Write/update tests
- [x] `make pr` passes
- [ ] Update documentation
- [ ] Verify transformed template deploys and application functions as expected

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by theBaffo** [comment]
Not sure why, but I got this error "UnboundLocalError: local variable 'patching' referenced before assignment", that's why the AppVeyor build failed.

**Comment by sriram-mv** [comment]
This looks well formed, Need to investigate on the appveyor failure.

**Comment by hawflau** [comment]
Hey @theBaffo it looks like you've defined `HostedZoneId` in `policy_templates.json`, but used `hostedZoneId` (lowercase h) in the tests. Can you update all `hostedZoneId` to `HostedZoneId` in the test input and outputs? It will pass the test then.

You can also run `make pr` before you submit the PR. It will run unit tests in your local machine.

**Comment by theBaffo** [comment]
> Hey @theBaffo it looks like you've defined `HostedZoneId` in `policy_templates.json`, but used `hostedZoneId` (lowercase h) in the tests. Can you update all `hostedZoneId` to `HostedZoneId` in the test input and outputs? It will pass the test then.
> 
> You can also run `make pr` before you submit the PR. It will run unit tests in your local machine.

Hi @hawflau!

Yes, my bad! I updated `hostedZoneId` to `HostedZoneId` as requested, and update the PR. It should be good to go now!

**Review by hawflau — APPROVED** [review]
LGTM!

File: samtranslator/policy_templates_data/policy_templates.json [hunk 1]
```diff
@@ -2334,6 +2334,34 @@
...
+      "Definition": {
+        "Statement": [
+          {
+            "Effect": "Allow",
+            "Action": [
+              "route53:ChangeResourceRecordSets"
+            ],
+            "Resource": {
+              "Fn::Sub": [
+                "arn:${AWS::Partition}:route53:${AWS::Region}:${AWS::AccountId}:hostedzone/${HostedZoneId}",
```
Comment (hawflau) [inline samtranslator/policy_templates_data/policy_templates.json:2354]: This resource arn does not seem to require Region and AccountId.
See: https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/r53-api-permissions-ref.html#required-permissions-resource-record-sets

**Review by hawflau — CHANGES_REQUESTED** [review]

File: samtranslator/policy_templates_data/policy_templates.json (hunk 1 above)
Comment (theBaffo) [reply samtranslator/policy_templates_data/policy_templates.json:2354]: Sorry, thought it also need Region and AccountId.

I'll change it to "arn:${AWS::Partition}:route53:::hostedzone/${HostedZoneId}", and updated the code/tests.

**Review by hawflau — APPROVED** [review]

File: samtranslator/policy_templates_data/policy_templates.json [hunk 2]
```diff
@@ -2334,6 +2334,34 @@
           }
         ]
       }
+    },
+    "Route53ChangeResourceRecordSetsPolicy": {
+      "Description": "Gives permission to change resource records sets in Route53",
```
Comment (hoffa) [inline samtranslator/policy_templates_data/policy_templates.json:2339]: nit:
```suggestion
      "Description": "Gives permission to change resource record sets in Route 53",
```

**Review by hoffa — APPROVED** [review]
LGTM 🙌 (Applied description nit above)

**Pull request merged**